src/sql/src/plan/statement/ddl.rs

```diff
@@ -5012,13 +5093,61 @@ pub fn describe_alter_object_swap(
 }
 
 pub fn plan_alter_object_swap(
-    _: &mut StatementContext,
-    _: AlterObjectSwapStatement,
+    scx: &mut StatementContext,
+    stmt: AlterObjectSwapStatement,
 ) -> Result<Plan, PlanError> {
-    Err(PlanError::Unsupported {
-        feature: "ALTER ... SWAP ...".to_string(),
-        issue_no: Some(12972),
-    })
+    scx.require_feature_flag(&vars::ENABLE_ALTER_SWAP)?;
+
+    let AlterObjectSwapStatement {
+        object_type,
+        name_a,
+        name_b,
+    } = stmt;
+    let object_type = object_type.into();
+
+    // We'll try 10 times to generate a temporary suffix.
+    let gen_temp_suffix = |check_fn: &dyn Fn(&str) -> bool| {
+        let mut attempts = 0;
+        let name_temp = loop {
+            attempts += 1;
+            if attempts > 10 {
+                tracing::warn!("Unable to generate temp id for swapping");
+                sql_bail!("unable to swap!");
```

Comment:
Same, uncovered

Reply:
It's quite hard to test this code path since it relies on the random suffix generation.